harness/tests/experiment/utils.py

```diff
@@ -35,21 +35,30 @@ def send(
     ) -> workload.Stream:
         self._training_metrics = []
         self._validation_metrics = []
+        batches_completed = 0
         interceptor = workload.WorkloadResponseInterceptor()
 
         for step_id in range(initial_step_id, initial_step_id + steps):
             stop_requested = False
-            yield from interceptor.send(workload.train_workload(step_id), [batches_per_step])
+            yield from interceptor.send(
+                workload.train_workload(
+                    step_id, batches_per_step=batches_per_step, batches_completed=batches_completed
+                ),
+                [batches_per_step],
```

Comment:
Probably if the workload is going to contain batches_per_step, we don't need to pass it as a secondary argument anymore, right?

Also, very optional hot-take: should we rename `batches_per_step` of train_workload to something that sounds like it might change from step to step?

Reply:
Yeah, I was worried about (putting off) digging into what else removing that arg would do, but it's totally redundant now so I'll take a hack at removing it.

Also, I agree, `batches_this_step` or `current_batches`..?